src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -1252,19 +1254,21 @@ public synchronized MergeThread getMergeThread(
   }
 
   synchronized boolean getBooleanSetting(String name, boolean val) {
```

Comment:
The synchronization on these get methods will need to be removed.

The locking order is still
commit: IndexState -> ShardState
startPrimary: ShardState -> IndexState
It is the reversal of lock ordering that leaves the system open to deadlock.

Reply:
Alright, removed the synchronization of the "get setting" methods